src/main/java/uk/gov/pay/ledger/agreement/resource/AgreementResource.java

```diff
@@ -40,12 +47,32 @@ public AgreementSearchResponse search(@Valid @BeanParam AgreementSearchParams qu
         return agreementService.searchAgreements(searchParams, uriInfo);
     }
 
+    private boolean databaseProjectionSnapshotIsBehindEventStream(AgreementEntity agreementEntity, EventDigest eventDigest) {
+        return agreementEntity.getEventCount() < eventDigest.getEventCount();
+    }
+
     @Path("/{agreementExternalId}")
     @GET
     @Timed
     public Agreement get(@PathParam("agreementExternalId") String agreementExternalId,
+                         @HeaderParam(HEADER_PARAM_X_CONSISTENT) Boolean isConsistent,
                          @Context UriInfo uriInfo) {
-        return agreementService.findAgreement(agreementExternalId)
-                .orElseThrow(() -> new WebApplicationException(Response.Status.NOT_FOUND));
+        if (Boolean.TRUE.equals(isConsistent)) {
+            try {
+                var eventDigest = eventService.getEventDigestForResource(agreementExternalId);
+
+                return agreementService.findAgreementEntity(agreementExternalId)
+                        .filter(agreementEntity -> !databaseProjectionSnapshotIsBehindEventStream(agreementEntity, eventDigest))
```

Comment:
The negation of this conditional make it a bit hard to follow. Either:
- change the filter method so the logic is reversed  (e.g. `databaseProjectionSnapshotIsUpToDateWithEventStream(…)`)
- keep it as `databaseProjectionSnapshotIsBehindEventStream(…)` but don’t negate it and swap the `.map(Agreement::from)` and `Agreement.from(agreementService.projectAgreement(eventDigest)))` bits